README.md

```diff
@@ -1,318 +1,403 @@
 ```
 
-or
+#### Usage
 
-```yaml
-- name: Send custom JSON data to Slack workflow
-  id: slack
-  uses: slackapi/slack-github-action@v1.27.0
-  with:
-    # This data can be any valid JSON from a previous step in the GitHub Action
-    payload: |
-      {
-        "key": "value",
-        "foo": "bar"
-      }
-  env:
-    SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
-```
+Add this Action as a [step][job-step] to your project's GitHub Action workflow
+file with the configurations you want.
 
-or
+##### Sending values from the GitHub default context
 
-> If the `payload` is provided it will take preference over `payload-file-path`
+In the example below, no payload input values are being provided so values from
+the [GitHub context][github-context] specific to the job are used:
 
 ```yaml
-- name: Send custom JSON data to Slack workflow
-  id: slack
-  uses: slackapi/slack-github-action@v1.27.0
+- name: Send GitHub Action data to a Slack workflow
+  uses: slackapi/slack-github-action@v2-development
   with:
-    payload-file-path: "./payload-slack-content.json"
-  env:
-    SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+    payload-delimiter: "_"
+    webhook: ${{ secrets.SLACK_WEBHOOK_URL }}
+    webhook-type: webhook-trigger
 ```
 
-> To send the payload file JSON as is, without replacing templated values with
-> `github.context` or `github.env`, set `payload-file-path-parsed` to `false`.
-> Default: `true`.
+While also using `payload-delimiter` the payload is flattened and stringified to
+match the webhook input format expected by Workflow Builder.
+
+##### Providing parsed payload information as strings
+
+Provided input values for payload information are sent to the webhook URL after
+the job is started:
 
 ```yaml
 - name: Send custom JSON data to Slack workflow
-  id: slack
-  uses: slackapi/slack-github-action@v1.27.0
+  uses: slackapi/slack-github-action@v2-development
   with:
-    payload-file-path: "./payload-slack-content.json"
-    payload-file-path-parsed: false
-  env:
-    SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+    webhook: ${{ secrets.SLACK_WEBHOOK_URL }}
+    webhook-type: webhook-trigger
+    payload: |
+      status: "${{ job.status }}"
+      option: "false"
 ```
 
-### Technique 2: Slack App
+### Technique 2: Slack API method
 
-By creating a new Slack app or using an existing one, this approach allows your GitHub Actions job to post a message in a Slack channel or direct message by utilizing the [chat.postMessage](https://api.slack.com/methods/chat.postMessage) API method. Using this approach you can instantly post a message without setting up Slack workflows.
+A bot token or user token or [token of some other kind][tokens] can be used to
+call one of many [Slack API methods][methods]! This includes
+[`chat.postMessage`][chat.postMessage] and the official `@slack/web-api`
+implemention of [`files.uploadV2`][files.uploadV2].
```

Comment:
I tried to bring this spirit to [the example section](https://github.com/slackapi/slack-github-action/tree/v2-development?tab=readme-ov-file#uploading-a-file) for uploading a file since it adds excellent context and because it was seeming strange to dive into these details right away! 🙏